### ISSUE TITLE
Infinite recursion in "breakCycle", resolving a prefixed!module, when using a second "require.config" call

### ISSUE THREAD

**Issue by dcleao** [jrburke/requirejs#1343]
This error does not occur if all the configuration is set once, before any require calls.
If however, part of the configuration is only provided later, through a second "require.config" call, then when checking for cycles, `breakCycle` enters in "infinite recursion".
The reason seems to be that somehow, requirejs got to believe that a plugin depends on itself, and as far as I know, it does not (and the proof should be that the same configuration when all given at start works successfully).

As far as I could tell, a module like "prefixed!realModule" should gain the following two in its depMaps property:
* prefixed
* realModule

In my case, however, it is somehow gaining:
* prefixed
* prefixed!realModule

which leads to the cycle. The id properties of the modules have an "_unnormalizedN" suffix in them.

My setup is somewhat like this:

**Configuration file (I)**
```javascript
require.config({
    paths: {
        "app": "/some/location",
        "wrap": "/some/other/location"
    },
    map: {
       "*": {
           "jquery": "wrap!app/lib/jquery"
       }
    },
    config: {
       // Dynamic shimming by wrapping
        wrap: {
             "jquery": {exports: "$"},
             "jquery.ui": {
                 exports: "$",
                 deps: {
                    "jquery": "jQuery" // moduleId: parameterName, for the wrapper define function call
                }
             }
        }
    }
});
```

**Plugin file (I)**
```javascript
define(["wrap!app/lib/jquery.ui"], function($) {
    // ....
});
```

Everything works if the config is performed only once, like above. It fails, however, if it is performed in two stages:

**Configuration file (II)**
```javascript
require.config({
    paths: {
        "app": "/some/location",
        "wrap": "/some/other/location"
    },
    map: {
       "*": {
           "jquery": "wrap!app/lib/jquery"
       }
    },
    config: {
       // Dynamic shimming by wrapping
        wrap: {
             "jquery": {exports: "$"}
        }
    }
});
```

**Plugin file (II)**

```javascript
require.config({
    config: {
       // Dynamic shimming by wrapping
        wrap: {
             "jquery.ui": {
                 exports: "$",
                 deps: {
                    "jquery": "jQuery" // moduleId: parameterName, for the wrapper define function call
                }
             }
        }
    }
});

define(["wrap!app/lib/jquery.ui"], function($) {
    // ....
});
```

The `breakCycle` method then recurses infinitely following the `"wrap!app/lib/jquery_unnormalized3"` module.

I've checked that the "wrap" plugin is receiving the exact same configuration in both cases. The problem seems to be something that is done to the module objects on the second config call, that somehow messes their internal structure, causing it to say that a module depends on itself.

The "wrap" plugin, mentioned above is [this one](https://github.com/webdetails/cdf/blob/master/cdf-core/cdf/js-lib/require-amd/nonamd.js). It does prevent jQuery from registering under the name `"jquery"`.
I am using the latest requirejs version (2.1.17).

Do you have any idea why this is happening? Any workaround I can do to solve the problem? Do you need anymore information? Thanks!

**Comment by dcleao** [comment]
Hi, I tried to reproduce the bug, as described, in a detached sample, and it wasn't revealing itself... Then I went on adding more and more of the existing setup et voila, I got it. It only manifests itself if there is another module, besides "jquery.ui", that also depends on "jquery". I've setup a sample that you can use to reproduce, [here](https://www.dropbox.com/s/3sd3iwxc68q0bkv/breakCycleBug.zip?dl=0).

**Comment by dcleao** [comment]
Hi @jrburke, **please** give me some advice on this one.

**Comment by jrburke** [comment]
Thank you very much for the test case, that really helped!

It was a bug in requirejs. I fixed it and will be part of the 2.1.18 release. I am not sure when it will come out yet, hopefully on the order of weeks vs months, but if you need a fix in the meantime, you can look at the diff in the fix on this bug, it is quite a small fix.

**Comment by dcleao** [comment]
Thank you!
